Homework1_part2

Patsiomkina888 opened the pull request:
Выполнены задачи из первой домашней работы : 

операторы if, while, exception

Patsiomkina888 committed:
updated .gitignore

Patsiomkina888 committed:
implemented hometask "bot"

Patsiomkina888 committed:
Homework1_part_2_ex.1-7

Chyrets commented on 2_if2.py:
```diff
@@ -15,12 +15,26 @@
+                return 0
+
+    else:
+        return 0
```
тут у тебя получилось слишком много вложенных блоков, я бы на твоем месте пошел от обратного: сделал два отдельных блока if, в первом блоке проверил бы что str1 и/или str2 не являются строками, а во втором блоке уже проверял их длины

Patsiomkina888 replied:
исправила

Chyrets commented on 3_for.py:
```diff
@@ -14,14 +14,62 @@
+    return round(sum / len(list_of_items), 2)
+
+
+def sum(list_of_numbers):
```
в питоне уже есть встроенная функция sum и не желательно создавать свою функцию с таким же названием

Chyrets commented on 5_while2.py:
```diff
@@ -15,13 +15,23 @@
+        question = input("Задай свой вопрос ")
+        if question in answers_dict:
+            print(answers_dict.get(question))
+            # break - если необходимо закнчить вопрос после того, как найден ответ в словаре
```
тут все таки лучше добавить brake, лучше не создавать бесконечных циклов

Chyrets commented on 8_ephem_bot.py:
```diff
@@ -16,39 +16,80 @@
+    user_text = update.message.text  # например, user_text = Mars
+    command_and_planet_list = user_text.split()
+
+    planet_name = command_and_planet_list[1].lower()
```
вот тут было бы неплохо добавить блок try/except, так как тут может упасть ошибка IndexError

Patsiomkina888 committed:
- добавила обработку ошибки в бот
- добавила brake в 5_while2
- изменила sum в 3_for
- изменила вложенность в 2_if2

Chyrets commented on 3_for.py:
```diff
@@ -20,47 +20,29 @@
-def sum(list_of_numbers):
-    sum = 0
+def prouct_sum(list_of_numbers):
+    prouct_sum = 0
```
не критично, но лучше не называть переменные так же как и функции
